x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/response/googlevertexai/GoogleVertexAiRerankResponseEntity.java

```diff
@@ -109,14 +109,19 @@ private static List<RankedDocsResults.RankedDoc> doParse(XContentParser parser)
                 throw new IllegalStateException(format(FAILED_TO_FIND_FIELD_TEMPLATE, RankedDoc.SCORE.getPreferredName()));
             }
 
-            return new RankedDocsResults.RankedDoc(index, parsedRankedDoc.score, parsedRankedDoc.content);
+            if (parsedRankedDoc.id == null) {
+                throw new IllegalStateException(format(FAILED_TO_FIND_FIELD_TEMPLATE, RankedDoc.ID.getPreferredName()));
+            }
+
+            return new RankedDocsResults.RankedDoc(Integer.parseInt(parsedRankedDoc.id), parsedRankedDoc.score, parsedRankedDoc.content);
```

Comment:
How about we catch the `NumberFormatException` that could be thrown by `parseInt` and provide a more friendly message about what went wrong?

Maybe add a test for an invalid integer too.

Reply:
Updated in https://github.com/elastic/elasticsearch/pull/117287/commits/68e2721e5d20f3efdc486a4f257b0256b68feaa7